File: clojure/p001.clj
; Solution to Project Euler problem 1
; by Alex Reichert

; https://github.com/reichert621/Project-Euler-solutions

(defn factor-of? [f n]
  (zero? (rem n f)))

(defn sum [s]
  (reduce + s))

(defn has-factors-in? [n coll]
  (some #(factor-of? % n) coll))

(defn compute []
  (sum (filter #(has-factors-in? % [3 5]) (range 1000))))

File: ruby/p001.rb
#
# Solution to Project Euler problem 1
# by Alex Reichert
#
# https://github.com/reichert621/Project-Euler-solutions
#

def compute
  (0...1000).reduce { |sum, n| (n % 3 == 0 || n % 5 == 0) ? sum + n : sum }
end
